### ISSUE TITLE
[ONNX ] export | tensors spread across devices cuda and cpu

### ISSUE THREAD

**Issue by miguelvalente** [pytorch/pytorch#81096]
### 🐛 Describe the bug

```python
model_inputs = model_inputs.cpu()
model = model.cpu()

# traced_module = torch.jit.script(model, model_inputs)
# print(traced_module.code)

torch.onnx.export(
        model,
        model_inputs,
        output_,
        verbose=True,
        input_names=input_names,
        output_names=output_names)
```

In this example you see the trace and how functional.py:2486 has tensors in cuda. 
```
  %247 : Float(requires_grad=0, device=cuda:0) = onnx::Constant[value={2}]() # /home/mvalente/.local/lib/python3.8/site-packages/torch/nn/functional.py:2486:0
  %248 : Float(16, *, 768, device=cpu) = onnx::Pow(%246, %247) # /home/mvalente/.local/lib/python3.8/site-packages/torch/nn/functional.py:2486:0
  %249 : Float(16, *, 1, device=cpu) = onnx::ReduceMean[axes=[-1]](%248) # /home/mvalente/.local/lib/python3.8/site-packages/torch/nn/functional.py:2486:0
  %250 : Float(requires_grad=0, device=cuda:0) = onnx::Constant[value={1e-05}]() # /home/mvalente/.local/lib/python3.8/site-packages/torch/nn/functional.py:2486:0
```
I even went as far as forcing tensors on CPU directly in ```functional.py```, and keep having the same kind of output. 
![image](https://user-images.githubusercontent.com/39302216/177947454-ffbb61b2-c930-4dc0-9e35-954e2b09526a.png)

On another note operations I had to change ``` torch.expand()``` due to the same issue, but after the tensors stayed on CPU. From what I figured, onnx does not support or does not deal well with some types of operations. But I'm at a loss with layernorm and with GELU.

To summarise, actions performed through ```F.gelu``` and ```F.layernorm()``` somehow keep make the tensors stay in 'cuda'.

PS:
I have also tried to run with inputs and model in cuda to no avail.

### Versions

> Collecting environment information...
> PyTorch version: 1.11.0+cu113
> Is debug build: False
> CUDA used to build PyTorch: 11.3
> ROCM used to build PyTorch: N/A
> 
> OS: Ubuntu 20.04.3 LTS (x86_64)
> GCC version: (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0
> Clang version: 10.0.0-4ubuntu1 
> CMake version: Could not collect
> Libc version: glibc-2.31
> 
> Python version: 3.8.10 (default, Mar 15 2022, 12:22:08)  [GCC 9.4.0] (64-bit runtime)
> Python platform: Linux-5.13.0-1031-aws-x86_64-with-glibc2.29
> Is CUDA available: True
> CUDA runtime version: 11.5.119
> GPU models and configuration: GPU 0: NVIDIA A10G
> Nvidia driver version: 510.73.05
> cuDNN version: Probably one of the following:
> /usr/lib/x86_64-linux-gnu/libcudnn.so.8.3.2
> /usr/lib/x86_64-linux-gnu/libcudnn_adv_infer.so.8.3.2
> /usr/lib/x86_64-linux-gnu/libcudnn_adv_train.so.8.3.2
> /usr/lib/x86_64-linux-gnu/libcudnn_cnn_infer.so.8.3.2
> /usr/lib/x86_64-linux-gnu/libcudnn_cnn_train.so.8.3.2
> /usr/lib/x86_64-linux-gnu/libcudnn_ops_infer.so.8.3.2
> /usr/lib/x86_64-linux-gnu/libcudnn_ops_train.so.8.3.2
> HIP runtime version: N/A
> MIOpen runtime version: N/A
> Is XNNPACK available: True
> 
> Versions of relevant libraries:
> [pip3] numpy==1.23.0
> [pip3] stam-pytorch==0.0.4
> [pip3] torch==1.11.0+cu113
> [pip3] torchaudio==0.11.0+cu113
> [pip3] torchvision==0.12.0+cu113
> [conda] Could not collect
>

**Comment by AllenTiTaiWang** [comment]
Hi @miguelvalente ,

Thanks for reporting the issue! 
These constants look like the parameters in torch.layernorm, and are not from inputs and models. Could you check if they are all cpu in torch IR? I wonder if they are actually used on CUDA. And also provide a code (model/input) that can reproduce this behavior.

**Comment by miguelvalente** [comment]
Thanks @AllenTiTaiWang  for taking a look.

Unfortunately I cannot share the details that you're asking for. The problem was solved by installing a version of PyTorch that is cpu.
```pip3 install torch torchvision torchaudio --extra-index-url https://download.pytorch.org/whl/cpu```
